### PR TITLE
LPS-194407 Validate ObjectDefinition ERC lenght

### PR DESCRIPTION
**Pull request by murilostodolni**
Related: https://liferay.atlassian.net/browse/LPS-194407

cc: @Artartur

### CODE REVIEW

**Comment by murilostodolni** [comment]
ci:test:sf

**Comment by murilostodolni** [comment]
ci:test:object

**Comment by murilostodolni** [comment]
ci:test:relevant

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 1]
```diff
@@ -7440,6 +7440,7 @@ for-multiple-domains,-separate-each-domain-by-a-comma=For multiple domains, sepa
 for-security-reasons-upload-field-repeatability-is-limited-the-limit-is-defined-in-x-system-settings-x=For security reasons, upload field repeatability is limited. The limit is defined in {0}System Settings{1}.
 for-x-y=For {0}, {1}.
 forbidden=Forbidden
+forbidden-unmodifiable-system-object-definition-external-reference-code=Forbidden unmodifiable system object definition external reference code.
```
[the remark this answers is not in the document]
Comment (murilostodolni) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:7443]: Will we change the current message error, let we need to change the tests as well. What do you think?

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/jaxrs/exception/mapper/ObjectActionNameExceptionMapper.java [hunk 2]
```diff
@@ -37,15 +38,29 @@ public class ObjectActionNameExceptionMapper
...
 		return new Problem(
 			JSONUtil.putAll(
 				JSONUtil.put(
 					"fieldName", "name"
 				).put(
-					"message",
-					_language.get(
-						_acceptLanguage.getPreferredLocale(),
-						objectActionNameException.getMessageKey())
+					"message", messageKey
```
[the remark this answers is not in the document]
Comment (murilostodolni) [reply modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/jaxrs/exception/mapper/ObjectActionNameExceptionMapper.java:63]: Done. Thank you

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/jaxrs/exception/mapper/ObjectDefinitionExternalReferenceCodeExceptionMapper.java [hunk 3]
```diff
@@ -33,7 +38,34 @@ protected Problem getProblem(
 		ObjectDefinitionExternalReferenceCodeException
 			objectDefinitionExternalReferenceCodeException) {
 
-		return new Problem(objectDefinitionExternalReferenceCodeException);
+		String messageKey =
+			objectDefinitionExternalReferenceCodeException.getMessageKey();
+
+		if (messageKey == null) {
+			messageKey =
+				objectDefinitionExternalReferenceCodeException.getMessage();
```
[the remark this answers is not in the document]
Comment (murilostodolni) [reply modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/jaxrs/exception/mapper/ObjectDefinitionExternalReferenceCodeExceptionMapper.java:46]: Yes, agreed.

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/jaxrs/exception/mapper/ObjectFolderNameExceptionMapper.java [hunk 4]
```diff
@@ -36,11 +36,24 @@ public class ObjectFolderNameExceptionMapper
...
+		}
+
 		return new Problem(
 			Response.Status.BAD_REQUEST,
-			_language.get(
-				_acceptLanguage.getPreferredLocale(),
-				objectFolderNameException.getMessageKey()));
+			_language.format(
+				_acceptLanguage.getPreferredLocale(), messageKey,
+				objectFolderNameException.getArguments()));
```
[the remark this answers is not in the document]
Comment (murilostodolni) [reply modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/jaxrs/exception/mapper/ObjectFolderNameExceptionMapper.java:56]: Agrred

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java [hunk 5]
```diff
@@ -1877,12 +1879,26 @@ private void _validateExternalReferenceCode(
...
 			throw new ObjectDefinitionExternalReferenceCodeException.
 				MustNotStartWithPrefix();
 		}
+
+		char[] externalReferenceCodeCharArrayCharArray =
+			externalReferenceCode.toCharArray();
+
+		if (externalReferenceCodeCharArrayCharArray.length > 75) {
+			throw new ObjectDefinitionExternalReferenceCodeException.
+				MustBeLessThan75Characters();
```
[the remark this answers is not in the document]
Comment (murilostodolni) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java:1900]: Agreed

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties (hunk 1 above)
[the remark this answers is not in the document]
Comment (carolmariaabb) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:7443]: I confess that I find the current message a little confusing, but let's keep it this way then. Thanks!

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:test:object

**Comment by murilostodolni** [comment]
Error semantic versioning is related, I will solve the semver in `object-api`

**Comment by murilostodolni** [comment]
Semver and rebase with master

**Comment by murilostodolni** [comment]
ci:test:sf

**Comment by murilostodolni** [comment]
ci:test:object

**Comment by murilostodolni** [comment]
ci:test:relevant

**Comment by murilostodolni** [comment]
ci:forward:force

**Pull request closed without merging**